docs/content/en/docs/concepts/executable-playbooks.md

```diff
@@ -0,0 +1,387 @@
+    "on_completion": "end--db937fc8-3a42-41cc-b828-ec2db212f425",
+    "type": "if-condition",
+    "condition": "__soarca_ssh_result__:value != '01-01-2025'",
+    "on_true": "action--7fe08053-3685-4d8c-bc0a-40efce75113e"
+}
+```
+
+SOARCA supports variable interpolation, which means that variables can be used inside the `condition` field, as seen in the example above.
+
+Similarly, CACAO specifies `while-condition` steps, whose `on_true` branch will be repeatedly executed until the condition evaluates to false. The execution limit is currently set to 10. If the condition is still true after 10 iterations, the step ends in a failure.
+
+The `parallel` step allows executing multiple branches (in parallel) specified in the `next_steps` field. At this time, the steps in `next_steps` are executed sequentially. Parallel execution is scheduled for a later release in SOARCA.
+
+Next, we explain variables in CACAO and SOARCA.
+
+### Variables
+
+The CACAO specification allows defining variables on the playbook level, as well as on the step level. Playbook variables are available throughout the playbook. Variables defined on the step level are available in that step, and in any step that executes in a sub-branch of an `if-condition`, `while-condition`, or `parallel` step.
+
+According to the CACAO spec, variable names should start and end with double underscores (`__`), but this is not enforced by SOARCA. Furthermore, the CACAO spec allows defining multiple types of variables (strings, ip-addresses, numbers), but at this time SOARCA will interpret every value as a string. The `constant` and `external` fields are also ignored.
+
+```json
+"playbook_variables": {
+    "__target_ip__": {
+      "type": "ipv4-addr",
+      "description": "IP address of target system",
+      "constant": false,
+      "external": true
+    }
+}
+```
+
+SOARCA supports the interpolation of variables in different strings. The specific string-based fields that support interpolation depend on the capability. In general, string interpolation is supported in the fields of agents, targets, and `command` fields.
+
+Variable interpolation happens at the last possible moment, which means that step-dependant variables can be used in agent and target definitions.
+
+Substitution is performed by replacing any occurrence of `[variable_name]:value` with the string `value` of that variable. Undefined variables are not replaced.
+
+### Action steps
+
+Within CACAO playbooks, `action` steps can define commands that are executed by an _agent_ against one or more _targets_. The agent and targets are referenced by ID. SOARCA selects the internal capability for handling the step by looking at the `type` and `name` of the agent. After selecting the proper capability, SOARCA will sequentially perform every command in the `commands` field for every target specified in `targets`. If any command fails to execute successfully, further execution is halted and the step is considered to have failed.
```

Comment:
"if any command fails to execute successfully" is relatively high-level.
I think that what we consider a failed execution is when our functions return err != nil, so a fail due to service error. This is in contrast with the interpretation of "fails to execute successfully" as in "the returned result is not what was expected", which is instead supposed to be handled at higher level with the on_success, on_failure.

We may want to make this distinction more explicit?

Reply:
These are mare more explicit in the docs on the capabilities, because what constitutes a failure depends on the capability.

There is nothing in the spec that suggest we should treat a connection failure differently than a 404 http response, or the same. So it's up to the implementer to make these decisions.

Reply:
Made this explicit in the if/while docs.